saleor/graphql/core/mutations.py

```diff
@@ -398,6 +399,14 @@ def handle_typed_errors(cls, errors: list, **extra):
             extra.update({cls._meta.error_type_field: errors})
         return cls(errors=errors, **extra)
 
+    @staticmethod
+    def call_event(func_obj):
```

Comment:
What do you think about this function throwing an error when it's not called in a transaction? Then we could also rename it to "on_transaction_commit" and use it as a general-purpose function that calls different things (not only events) and ensures that the call always happen after the transaction is done? CC @patrys

Reply:
Wouldn't be more readable if instead of:
```
       cls.call_event(
             lambda: info.context.plugins.sale_created(
                 instance,
                 current_catalogue,
            )
        )
```
We would have:
```
cls.call_event(info.context.plugins.sale_created, instance, current_catalogue)
```
And inside `call_event`:
```
def call_event(func_obj, *fun_args):
    ...
     if connection.in_atomic_block:
         transaction.on_commit(func_obj, *fun_args)
     else:
         func_obj(*fun_args)
```
Thanks to this, we don't need to remember about adding `lambda`, and I think the rest of the code will be more readable.

Reply:
@maarcingebala As I remember we didn't want all the webhook events to be called in transaction. But we wanted to make sure that if they  were - they are called `on_commit`.

@korycins I like this idea